samples/cpp/speech_sample/main.cpp

```diff
@@ -286,161 +298,182 @@ int main(int argc, char* argv[]) {
+                } else if (exReferenceScoresFile == "npz") {
+                    fileReferenceScores = &numpyFile;
+                } else {
+                    throw std::logic_error("Invalid Reference Scores file");
                 }
-                ptrScores.resize(numFrames * numScoresPerFrame * sizeof(float));
-                if (!FLAGS_r.empty()) {
-                    /** Read file with reference scores **/
-                    BaseFile* fileReferenceScores;
-                    auto exReferenceScoresFile = fileExt(FLAGS_r);
-                    if (exReferenceScoresFile == "ark") {
-                        fileReferenceScores = &arkFile;
-                    } else if (exReferenceScoresFile == "npz") {
-                        fileReferenceScores = &numpyFile;
-                    } else {
-                        throw std::logic_error("Invalid Reference Scores file");
-                    }
-                    std::string refUtteranceName;
-                    fileReferenceScores->get_file_info(reference_name_files[next_output].c_str(),
+                for (size_t next_output = 0; next_output < count_file; next_output++) {
+                    if (fileReferenceScores != nullptr) {
+                        fileReferenceScores->get_file_info(reference_name_files[next_output].c_str(),
+                                                           utteranceIndex,
+                                                           &n,
+                                                           &numBytesReferenceScoreThisUtterance);
+                        ptrReferenceScores.resize(numBytesReferenceScoreThisUtterance);
+                        fileReferenceScores->load_file(reference_name_files[next_output].c_str(),
                                                        utteranceIndex,
-                                                       &n,
-                                                       &numBytesReferenceScoreThisUtterance);
-                    ptrReferenceScores.resize(numBytesReferenceScoreThisUtterance);
-                    fileReferenceScores->load_file(reference_name_files[next_output].c_str(),
-                                                   utteranceIndex,
-                                                   refUtteranceName,
-                                                   ptrReferenceScores,
-                                                   &numFramesReference,
-                                                   &numFrameElementsReference,
-                                                   &numBytesPerElementReference);
+                                                       refUtteranceName,
+                                                       ptrReferenceScores,
+                                                       &numFramesReference,
+                                                       &numFrameElementsReference,
+                                                       &numBytesPerElementReference);
```

Comment:
We need a test for model with several outputs

Reply:
Now we make a comparison with reference for each output.